cityofaustin/atd-moped
# Refactor component <> feature relationship from many-to-many to many-to-one

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/8389

Todo:
- [x] Test/update: links to signals from dashboard, project summary
- [x] Create a migration to backfill existing projects
- [x] Test that fetching the complete geometry from agol is working
- [x] Think about the use of `PROJECT_EXTENT_ID` in this code: i'd like to change up the way we use this property, but can leave it out of scope. Chia took care of all the immediate needs in #546 
- [x] Test/update: Knack sync: Holding off until #522 can be incorporated
- [ ] Not doing this: Delete all projects in staging - will do before merging [8157-new-tile-endpoint](https://github.com/cityofaustin/atd-moped/tree/8157-new-tile-endpoint) to main

## Summary of changes

- deprecate `moped_proj_features_components` associative table
- remove fk relationship between `moped_proj_features` and `moped_project`
- deprecate `COMPONENT_DETAILS_QUERY`, and move the source for moped_components and moped_subcomonents up one level in the tree.
- refactor the data munging involved with parsing and packaging up mutation payloads, which are simpler now that we don't have the associative table, moped_proj_features_components.
- hookify and move up the tree the functions that index moped_components and moped_subcomponents

## Testing
**URL to test:** [Moped Test](https://moped-test.austinmobility.io/moped/)

**Steps to test:**
Make sure there are no UI regressions. Create, edit, and delete lots of components, with subcomponents.
Use **multiple features** per component. Including:
- A signal component created through the new project workflow
- A selected point(s) component (create/edit/delete)
- A selected line(s) component (create/edit/delete)
- A drawn line(s) component (create/edit/delete)
- A drawn point(s) component (create/edit/delete)
- A mix of drawn/selected points
- A mix of drawn/selected lines
- Verify the summary map renders correctly
- Verify the map tab renders correctly
- Verify that the signals dashboard (and signal links) renders correctly
- Verify that the project summary (and signal links) renders correctly

---
#### Ship list
